src/protocol/primitives/compact-nullable-bytes.ts

```diff
@@ -0,0 +1,24 @@
+import type { ReadBuffer, Serializable, WriteBuffer } from '../serialization';
+import { UVarInt } from './uvarint';
+
+export class CompactNullableBytes implements Serializable {
+  constructor(public readonly value: Buffer | null) {}
+
+  public static deserialize(buffer: ReadBuffer): CompactNullableBytes {
+    const length = UVarInt.deserialize(buffer);
+    if (length.value === 0) {
+      return new CompactNullableBytes(null);
+    }
+
+    return new CompactNullableBytes(buffer.readBuffer(length.value));
```

Comment:
The bytes to read should be length-1, as declared in the protocol specification: 

> COMPACT_NULLABLE_BYTES: [...]  First the length **N+1** is given as an UNSIGNED_VARINT. Then N bytes follow. A null object is represented with a length of 0.